synapse/handlers/room.py

```diff
@@ -171,6 +174,15 @@ async def upgrade_room(
     async def _upgrade_room(
         self, requester: Requester, old_room_id: str, new_version: RoomVersion
     ):
+        """
+        Args:
+            requester: the user requesting the upgrade
+            old_room_id: the id of the room to be replaced
+            new_versions: the version to upgrade the room to
+
+        Raises:
+            ShadowBanError if the requester is shadow-banned.
```

Comment:
I wonder if this should catch here and just create a new room for the shadow-banned person instead of continuing to raise `ShadowBanError`?

Reply:
Possibly, or just no-op. I think it'll be a pretty rare case given you need to be a mod to upgrade a room.

Reply:
Alright. This does get caught at the REST layer and a fake room ID is returned.